### PR TITLE
cli: `testsys status` update

### LINKED ISSUE #265: cli: rethink testsys status
We should do some design ideation on what we want a great `testsys status` experience to be and implement it. I envision going back to a columnar table, but let's discuss.

### PR DESCRIPTION
**Pull request by ecpullen**
**Issue number:**
Closes #265 

**Description of changes:**
Provides a clean output for `testsys status`.

**Testing done:**
```
┌────────────────────────┬────────────┬───────────┬────────┬─────────┬────────┐
│          Name          │    Type    │   State   │ Passed │ Skipped │ Failed │
├────────────────────────┼────────────┼───────────┼────────┼─────────┼────────┤
│       Controller       │ Controller │  Running  │        │         │        │
├────────────────────────┼────────────┼───────────┼────────┼─────────┼────────┤
│      testsys-test      │  Resource  │ completed │        │         │        │
├────────────────────────┼────────────┼───────────┼────────┼─────────┼────────┤
│ testsys-test-instances │  Resource  │ completed │        │         │        │
├────────────────────────┼────────────┼───────────┼────────┼─────────┼────────┤
│       ecs-smoke        │    Test    │  passed   │   1    │    0    │   0    │
└────────────────────────┴────────────┴───────────┴────────┴─────────┴────────┘
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by webern — CHANGES_REQUESTED** [review]
We've had an out-of-band discussion and arrived at something simpler. The `--wait` use case is fuzzy and our implementation is awkward. So we can simplify things significantly

- We should eliminate the `--wait` argument and functionality.
- `testsys status` always exists 0 (unless it has encountered an error in it's execution)
- We don't need to do anything with resetting the cursor.
- We can keep the work you've done to find the terminal size and use it to format a table, all of that still leads to nice output when `--json` is not used.

**Comment by ecpullen** [comment]
^ Removes the wait flag from the command and sets the command to always exit 0.

**Review by webern — APPROVED** [review]
😎

File: testsys/src/error.rs [hunk 1]
```diff
@@ -130,4 +130,10 @@ pub(crate) enum Error {
 
     #[snafu(display("Unable to write data: {}", source))]
     Write { source: std::io::Error },
+
+    #[snafu(display("Unable to get the cursor position: {}", source))]
+    Cursor { source: std::io::Error },
+
+    #[snafu(display("Unable to get the terminal size: {}", source))]
+    TerminalSize { source: std::io::Error },
```
Comment (etungsten) [inline testsys/src/error.rs]: Do we still need these errors variants?

File: testsys/tests/cli_test.rs [hunk 2]
```diff
@@ -180,19 +122,7 @@ async fn test_system() {
...
-        "--kubeconfig",
-        cluster.kubeconfig().to_str().unwrap(),
-        "status",
-        "-r",
-        "never-destroy",
-        "--wait",
-    ])
-    .timeout(POD_TIMEOUT);
-    cmd.assert().success();
+    tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
```
Comment (etungsten) [inline testsys/tests/cli_test.rs]: nit: This seems extraneous if we're waiting later in the loop anyways, could wait first in the loop before doing the assertion.

File: testsys/src/status.rs [hunk 3]
```diff
@@ -340,3 +298,127 @@ impl Display for ResourceResult {
...
+            min_state_width: 5,
+            min_passed_width: 6,
+            min_skipped_width: 7,
+            min_failed_width: 6,
+        }
+    }
+}
+
+impl Results {
+    fn draw(&self) {
```
Comment (etungsten) [inline testsys/src/status.rs:346]: Good job figuring this out! But it almost makes me wonder if there are other crates out there we can use to do this.

**Review by etungsten — APPROVED** [review]

**Pull request merged**